src/components/Tag/Tag.tsx

```diff
@@ -10,40 +10,73 @@ import styles from './Tag.scss';
 
 export interface NonMutuallyExclusiveProps {
   /** Content to display in the tag */
-  children?: string;
+  children?: string | React.ReactElement;
   /** Disables the tag  */
   disabled?: boolean;
-  /** Callback when tag is clicked or keypressed. Renders without remove button when set. */
+  /** Callback when tag is clicked or keypressed. Renders without remove button or url when set. */
   onClick?(): void;
   /** Callback when remove button is clicked or keypressed. */
   onRemove?(): void;
+  /** A string to use when tag has more than textual content */
+  title?: string;
+  /** Url to navigate to when tag is clicked or keypressed. */
+  url?: string;
 }
 
 export type TagProps = NonMutuallyExclusiveProps &
   (
-    | {onClick?(): void; onRemove?: undefined}
-    | {onClick?: undefined; onRemove?(): void}
+    | {onClick?(): void; onRemove?: undefined; url?: undefined}
+    | {onClick?: undefined; onRemove?(): void; url?: string}
   );
 
-export function Tag({children, disabled = false, onClick, onRemove}: TagProps) {
+export function Tag({
+  children,
+  disabled = false,
+  onClick,
+  onRemove,
+  title = '',
```

Comment:
Should we leave out a default value? Right now we can render a title without an content, but if we left it as undefined react would omit it

Reply:
Hey @AndrewMusgrave, I tried this (turns out already tried it before) and the reason I left it as empty was because down there we have

```
const ariaLabel = i18n.translate('Polaris.Tag.ariaLabel', {
  children: tagTitle,
});
```

So, `children` expects `string | number`. If `title` can be `undefined`, then we'd need additional logic for that as well, which might be fine... what do you think?